Assets/Scripts/BossInformation.cs

```diff
@@ -0,0 +1,29 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BossInformation : MonoBehaviour
```

Comment:
```suggestion
public class BossImmunityInformation : MonoBehaviour
```
The name `BossInformation` is a bit generic. How about this alternative?

And have you thought about combining this script with `MinionSpawnerController` and either applying the combined script to the Boss prefab, or to a child object of the Boss prefab?
That could even be the object that will have the sprite renderer for the shield in the near future - just a thought.